src/gui/tree-item.cc

```diff
@@ -36,6 +36,9 @@ namespace gepetto {
       setEditable(false);
       connect (&vmMapper_, SIGNAL (mapped (QString)), SLOT(setViewingMode(QString)));
       connect (&vizMapper_, SIGNAL (mapped (QString)), SLOT(setVisibilityMode(QString)));
+      
+      const std::string & name = node->getID();
+      QStandardItem::setText(name.substr(name.find_last_of("/") + 1).c_str());
```

Comment:
I think this fails if `name` does not contain `'/'`. From the doc of `std::string::find_last_of`:
```
If no matches are found, the function returns string::npos.
```

Reply:
`string::npos` is in fact equal to -1, so it should work too.

Reply:
and `substr` accept a negative input ?!? It isn't mentioned in the doc... Isn't it a bit hazardous to use this ?

Reply:
It won't be negative but zero.

Reply:
I checked the code of `substr`. It calls the constructor `string (string, pos, length)`:
```c++
basic_string(const basic_string& __str, size_type __pos,
		   size_type __n = npos)
      : _M_dataplus(_M_local_data())
      {
	const _CharT* __start = __str._M_data()
	  + __str._M_check(__pos, "basic_string::basic_string");
	_M_construct(__start, __start + __str._M_limit(__pos, __n));
      }
```
so I think `__start <- __str._M_data() - size_t(pos)`. So the value it gets should be the conversion of `npos` to an unsigned integral type.

Reply:
From [the doc of npos](http://www.cplusplus.com/reference/string/string/npos/):

This constant is defined with a value of -1, which because size_t is an unsigned integral type, it is the largest possible representable value for this type.

Reply:
Sorry for the remarks. I understand just now why it works. `size_t(-1) + 1` equals 0 by overflow...